client/src/pages/repo-browser/RepositoryTile.tsx

```diff
@@ -15,13 +15,27 @@ export const RepositoryTile = (props) => {
   const [isDisabled, setIsDisabled] = useState(false);
   const [isWarning, setIsWarning] = useState(false);
   const [isError, setIsError] = useState(false);
+  const [isPublic, setIsPublic] = useState(false);
   const [dayDifference, setDayDifference] = useState<number>();
   const [expires, setExpires] = useState();
   const [writeableBool, setWriteableBool] = useState<any>();
 
   useEffect(() => {
-    const tempExpires = sasToken.slice(sasToken.search('se')).split('&')[0].slice(3, 13); // YEAR-MONTH-DAY
-    const writeable = sasToken.slice(sasToken.search('sp')).split('&')[0].includes('w'); // boolean
+    let writeable = false;
+    if (sasToken == '') {
+      setIsPublic(true);
```

Comment:
you could just use if(!!sastoken) and avoid the else

Reply:
still need IsPublic flag